### DISCUSSION TITLE
Implementation of xxHash - Extremely fast hash algorithm

### DISCUSSION THREAD

**Discussion by RalphMichel** [doublecmd/doublecmd#1779] [Suggestions]
First of all, a thousand thanks for this wonderful tool. 

Since I work a lot with very large files and provide them with checksums, it would be fantastic if xxHash were also implemented in addition to the numerous hash algorithms. Compared to other algorithms, xxHash is extremely fast and secure.

https://github.com/Cyan4973/xxHash

**Comment by Skif-off** [comment]
> xxHash is extremely fast and **_secure_**

What do you mean? xxHash is a non-cryptographic hash algorithm.

**Reply by RalphMichel** [reply]
that is indeed wrong. sorry for that expression. i was thinking more of the quality and not the security.

**Comment by alexx2000** [comment]
Blake3 is a good enough for large files on computers with modern x86_64 processors. It has a speed which is comparable with NVMe drives.

Theoretically I can add XXH128  function. I guess a fewer bits make no sense for large files.

**Reply by RalphMichel** [reply]
Ok, to be honest, I've never used Blake3. I will test it now. Thank you!

**Comment by vivadavid** [comment]
Hi,

I also wanted to request adding support for the xxh3-64 algorithm. I know it's a non-cryptographic hash, but I just need something to check that my files aren't corrupted.

So far, I'm using Blake3, but I did some testing a while ago, and xxh3-64 turned out to be faster. Teracopy has recently adopted xxh3-64 as the default hash.

Thanks in advance!

**Comment by alexx2000** [comment]
Hi, a speed difference between xxh3-64 and xxh3-128 are small so I will prefer to add xxh3-128.

**Reply by vivadavid** [reply]
> Hi, a speed difference between xxh3-64 and xxh3-128 are small so I will prefer to add xxh3-128.

Hi, I'm not an expert on the subject, but I think that xxh3-128 is safer, but noticeably slower. I may have done some testing in the past, but I didn't record the results.

In any case, I wonder if it'd be very difficult to add both hashes. Personally, xxh3-64 is the one I'd be using, and the one I'm already using in Teracopy, where it's the hash by defaut.

**Comment by vivadavid** [comment]
Hi,

Just to support my claims about speed, I've done some testing. Using Teracopy 3.17, I've calculated the checksum of a folder contaning 5,23 GB and 3533 files.

This is the time it took:

- xxh3-64: 12 seconds.
- xxh3-128: 46 seconds.

**Comment by alexx2000** [comment]
I test using official `xxhsum.exe` tool, a speed difference about 2,5%:
```
XXH3-64                      :     102400 ->   205894 it/s (20106.8 MB/s)
XXH3-128                     :     102400 ->   200695 it/s (19599.2 MB/s)
```
